admin-auth/src/test/kotlin/com/epam/drill/admin/auth/OAuthServiceTest.kt

```diff
@@ -52,39 +50,42 @@ class OAuthServiceTest {
     }
 
     @Test
-    fun `given new OAuth2 principal, signInThroughOAuth must create new user`() = runBlocking {
-        val testUsername = "some-username"
-        val testRole = "user"
-        val testAccessToken = "test-access-token"
-
-        val httpClient = mockHttpClient(
-            "/userInfoUrl" to userInfoResponse(testAccessToken, testUsername, testRole)
-        )
-        val oauthService = OAuthServiceImpl(httpClient, OAuthConfig(config), userRepository)
-        whenever(userRepository.findByUsername(testUsername)).thenReturn(null)
-        whenever(userRepository.create(any())).thenReturn(1)
-
-        val userInfo = oauthService.signInThroughOAuth(withPrincipal(testAccessToken))
-        verify(userRepository).create(any())
-        assertEquals(testUsername, userInfo.username)
-        assertTrue(testRole.equals(userInfo.role.name, true))
-    }
+    fun `given user that is authenticated through the OAuth2 first time, signInThroughOAuth must create new user`() =
+        runBlocking {
+            val testUsername = "some-username"
+            val testRole = "user"
+            val testAccessToken = "test-access-token"
+
+            val httpClient = mockHttpClient(
+                "/userInfoUrl" shouldRespond assertTokenAndRespondSuccess(testAccessToken, testUsername, testRole)
+            )
+            val oauthService = OAuthServiceImpl(httpClient, OAuthConfig(config), userRepository)
+            whenever(userRepository.findByUsername(testUsername)).thenReturn(null)
+            whenever(userRepository.create(any())).thenReturn(1)
+
+            val userInfo = oauthService.signInThroughOAuth(withPrincipal(testAccessToken))
+            verify(userRepository).create(any())
+            assertEquals(testUsername, userInfo.username)
+            assertTrue(testRole.equals(userInfo.role.name, true))
+        }
 
     @Test
-    fun `given OAuth2 principal that was already logged in, signInThroughOAuth must update user role`() = runBlocking {
+    fun `given user that is authenticated through OAuth2 again, signInThroughOAuth must update user`() = runBlocking {
```

Comment:
1. `given existing user, authenticating with OAuth2, signInThroughOAuth must update user entity with data from "user-info"`
2. _tbd_ splitting test case

Reply:
only change name

Reply:
Will be done in user mapping branch